### ISSUE TITLE
Connect via UDP

### ISSUE THREAD

**Issue by MatzHilven** [felixge/node-graphite#19]
Is it possible to connect to carbon via UDP? (using this client of course)

**Comment by Serphentas** [comment]
I believe you can, but by default `carbon.conf` has that feature disabled:

```
ENABLE_UDP_LISTENER = False
UDP_RECEIVER_INTERFACE = 0.0.0.0
UDP_RECEIVER_PORT = 2003
```

However, this client uses LazySocket that [calls](https://github.com/felixge/node-lazy-socket/blob/master/lib/LazySocket.js#L68) for a normal [Socket](https://nodejs.org/api/net.html#netcreateconnection) creation which is only TCP. Support for UDP would require a PR at this stage.

Are you able to submit such a change ? If not, I can look into it.

**Comment by MatzHilven** [comment]
I'm not very experienced in js, but it looks like the `net` module has only got support for TCP clients. Seems like `dgram` would be good for this job?

**Comment by Serphentas** [comment]
That is correct, it's just a matter of calling the right type of socket within LazySocket, but this is outside the scope of this repo and I can only approve PRs here. Thus, I suggest we implement UDP support without dependencies as it's pretty straightforward.

I will think of a first draft to then use for tests and ping you when it's ready.

**Comment by MatzHilven** [comment]
Perfect, thanks!

**Comment by Serphentas** [comment]
I just pushed the initial version with UDP support on the [feat-udp](https://github.com/felixge/node-graphite/tree/feat-udp) branch. Can you pull it and try locally ?

You will also need to change the connect string: replace `plaintext` with `udp`. See README for an example.

**Comment by Serphentas** [comment]
Any updates @MatzHilven ? If it works well, I will merge into master.

**Comment by MatzHilven** [comment]
I'll get you an answer this evening!

**Comment by Serphentas** [comment]
Shameless ping @MatzHilven :)

**Comment by MatzHilven** [comment]
Sorry for the delay!!
LGTM 😊

Also: https://github.com/felixge/node-graphite/pull/20#issuecomment-1811197602
